### PR TITLE
Xml join optimization

### PR DESCRIPTION
**Pull request by olenagerasimova**
Optimized `XmlMetaJoin` by getting rid of xml steams as requested in #200. New implementation works faster:
```
//buffered reader
[DEBUG] main com.artipie.rpm.meta.XmlMetaJoin: /tmp/junit18146090115461577170/repo/fast.filelists.xml and /tmp/junit18146090115461577170/repo/big.filelists.xml merged in 138ms
//xml stream
[DEBUG] main com.artipie.rpm.meta.XmlMetaJoin: /tmp/junit18146090115461577170/repo/stream.filelists.xml and /tmp/junit18146090115461577170/repo/big.filelists.xml merged in 15s
```
For now I've decided to keep xml steams implementation for tests.

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#196](https://github.com/artipie/rpm-adapter/issues/196) is assigned to @carlosmiranda[/z](https://www.0crat.com/u/carlosmiranda), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/e99f6839-04d6-4f39-8356-d6aec1352dac); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/main/java/com/artipie/rpm/meta/XmlMetaJoin.java [hunk 1]
```diff
@@ -79,8 +98,34 @@ public void merge(final Path target, final Path part) throws IOException {
...
+    }
+
+    /**
+     * Appends data from part to target.
+     * @param target Target
+     * @param part File to append
+     * @throws IOException On error
+     */
+    @SuppressWarnings({"PMD.PrematureDeclaration", "PMD.GuardLogStatement"})
+    public void fastMerge(final Path target, final Path part) throws IOException {
```
Comment (carlosmiranda) [inline src/main/java/com/artipie/rpm/meta/XmlMetaJoin.java:114]: @olenagerasimova is there any reason (e.g. backwards compatibility) why we need _both_ `merge` and `fastMerge`? Maybe we can deprecate the former or add a puzzle to remove it?

**Comment by olenagerasimova** [comment]
@carlosmiranda thanks, I've created issue for this PR and added todo to delete/move old method.

**Comment by g4s8** [comment]
@olenagerasimova please check broken pdd linter: https://github.com/artipie/rpm-adapter/pull/196/checks?check_run_id=705115261
Master build is green for pdd

**Comment by 0crat** [comment]
Job `gh:artipie/rpm-adapter#196` is not assigned, can't get performer

**Pull request merged**